resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/submit/submitsteps/initcontainer/BaseInitContainerStepSuite.scala

```diff
@@ -37,16 +36,28 @@ class BaseInitContainerStepSuite extends SparkFunSuite with BeforeAndAfter{
   private val FILES_DOWNLOAD_PATH = "/var/data/files"
   private val CONFIG_MAP_NAME = "config-map"
   private val CONFIG_MAP_KEY = "config-map-key"
+  private val POD_LABEL = Map("bootstrap" -> "true")
+  private val INIT_CONTAINER_NAME = "init-container"
+  private val DRIVER_CONTAINER_NAME = "driver-container"
 
   @Mock
   private var podAndInitContainerBootstrap : SparkPodInitContainerBootstrap = _
-  @Mock
-  private var podWithDetachedInitContainer : PodWithDetachedInitContainer = _
 
   before {
     MockitoAnnotations.initMocks(this)
     when(podAndInitContainerBootstrap.bootstrapInitContainerAndVolumes(
-      any[PodWithDetachedInitContainer])).thenReturn(podWithDetachedInitContainer)
+      any[PodWithDetachedInitContainer])).thenReturn(
+      PodWithDetachedInitContainer(
+        new PodBuilder()
```

Comment:
We probably want to use the "base" spec here too. See for example the test implementations of the submission steps in `ClientSuite`.